CRM/Utils/Mail/Incoming.php

```diff
@@ -344,13 +344,7 @@ public static function parseMailingObject(&$mail, $createContact = TRUE, $requir
     }
 
     $params['from'] = [];
-    self::parseAddress($mail->from, $field, $params['from'], $mail, $createContact);
-
-    // we definitely need a contact id for the from address
-    // if we dont have one, skip this email
-    if ($requireContact && empty($params['from']['id'])) {
```

Comment:
This check doesn't make sense in the new context as we don't know what the mapping from email fields to activity contacts is here. But it doesn't matter as this is handled in EmailProcessor at 210 anyways, since the condition can also come up in other ways (for instance, if $requireContact is not true, we still can't create an activity without a source contact).